Upload on sync fails for LocalChangeResourceReferenceEntity

Upload on sync fails here with 

```
(1) row value misused in "SELECT *
        FROM LocalChangeResourceReferenceEntity
        WHERE localChangeId = (?,?,?,?,?,?,?,?,?,?,?)"
        
java.util.concurrent.ExecutionException: android.database.sqlite.SQLiteException: row value misused (code 1 SQLITE_ERROR): , while compiling: SELECT *
        FROM LocalChangeResourceReferenceEntity
        WHERE localChangeId = (?,?,?,?,?,?,?,?,?,?,?)
	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:515)
	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:474)
	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:316)
	at androidx.work.impl.utils.SerialExecutorImpl$Task.run(SerialExecutorImpl.java:96)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: android.database.sqlite.SQLiteException: row value misused (code 1 SQLITE_ERROR): , while compiling: SELECT *
        FROM LocalChangeResourceReferenceEntity
        WHERE localChangeId = (?,?,?,?,?,?,?,?,?,?,?)
	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1069)
	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:673)
	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:62)
	at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1714)
	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1689)
	at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.query(FrameworkSQLiteDatabase.kt:156)
	at androidx.room.RoomDatabase.query(RoomDatabase.kt:490)
	at androidx.room.util.DBUtil.query(DBUtil.kt:75)
	at com.google.android.fhir.db.impl.dao.LocalChangeDao_Impl$20.call(LocalChangeDao_Impl.java:927)
	at com.google.android.fhir.db.impl.dao.LocalChangeDao_Impl$20.call(LocalChangeDao_Impl.java:924)
	at androidx.room.CoroutinesRoom$Companion$execute$4$job$1.invokeSuspend(CoroutinesRoom.kt:88)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	... 3 more

```

Guessing the query should probably use an `IN` instead of `=`

_Originally posted by @LZRS in https://github.com/google/android-fhir/pull/2442#discussion_r1531727947_

--- a/engine/src/androidTest/java/com/google/android/fhir/db/impl/dao/LocalChangeDaoTest.kt
+++ b/engine/src/androidTest/java/com/google/android/fhir/db/impl/dao/LocalChangeDaoTest.kt
@@ -1,5 +1,5 @@
 /*
- * Copyright 2023 Google LLC
+ * Copyright 2023-2024 Google LLC
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -35,6 +35,7 @@ import org.hl7.fhir.r4.model.CarePlan
 import org.hl7.fhir.r4.model.CodeableConcept
 import org.hl7.fhir.r4.model.Coding
 import org.hl7.fhir.r4.model.Enumerations
+import org.hl7.fhir.r4.model.Observation
 import org.hl7.fhir.r4.model.Patient
 import org.hl7.fhir.r4.model.Reference
 import org.junit.After
@@ -355,5 +356,45 @@ class LocalChangeDaoTest {
       .isEqualTo("activity.detail.performer")
     assertThat(localChangeResourceReferences[2].resourceReferenceValue)
       .isEqualTo(practitionerReference)
+  }
+
+  @Test
+  fun getReferencesForLocalChanges_should_return_all_changes(): Unit = runBlocking {
+    listOf(
+        Observation().apply {
+          id = "1"
+          subject = Reference("Patient/1")
+          encounter = Reference("Encounter/1")
+        },
+        Observation().apply {
+          id = "2"
+          subject = Reference("Patient/2")
+          encounter = Reference("Encounter/2")
+        },
+      )
+      .forEach {
+        localChangeDao.addInsert(
+          it,
+          UUID.randomUUID(),
+          Instant.now(),
+        )
+      }
+
+    // we need LocalChangeEntity.id (auto generated) to call getReferencesForLocalChanges api
+    val localChangesMap = localChangeDao.getAllLocalChanges().associateBy { it.resourceId }
+
+    // Calling the api for single local change
+    var result = localChangeDao.getReferencesForLocalChanges(listOf(localChangesMap["1"]!!.id))
+    assertThat(result).hasSize(2)
+    assertThat(result.map { it.resourceReferenceValue }).containsExactly("Patient/1", "Encounter/1")
+
+    // Calling the api for multiple local changes
+    result =
+      localChangeDao.getReferencesForLocalChanges(
+        listOf(localChangesMap["1"]!!.id, localChangesMap["2"]!!.id),
+      )
+    assertThat(result).hasSize(4)
+    assertThat(result.map { it.resourceReferenceValue })
+      .containsExactly("Patient/1", "Encounter/1", "Patient/2", "Encounter/2")
   }
 }

--- a/engine/src/main/java/com/google/android/fhir/db/impl/dao/LocalChangeDao.kt
+++ b/engine/src/main/java/com/google/android/fhir/db/impl/dao/LocalChangeDao.kt
@@ -352,7 +352,7 @@ internal abstract class LocalChangeDao {
     """
         SELECT *
         FROM LocalChangeResourceReferenceEntity
-        WHERE localChangeId = (:localChangeId)
+        WHERE localChangeId IN (:localChangeId)
     """,
   )
   abstract suspend fun getReferencesForLocalChanges(

